cheetah/accelerator.py

```diff
@@ -299,6 +307,354 @@ def defining_features(self) -> list[str]:
     def __repr__(self) -> str:
         return f"{self.__class__.__name__}(length={repr(self.length)})"
 
+class SpaceChargeKick(Element):
+    """
+    Simulates space charge effects on a beam.
+    :param length: Length of the element in meters.
+    :param grid_precision: Number of grid points in each dimension.
+    :param grid_dimensions: Dimensions of the grid as multiples of sigma of the beam.
+    :param name: Unique identifier of the element.
+    """
+
+    def __init__(
+        self,
+        length: Union[torch.Tensor, nn.Parameter],
+        nx: Union[torch.Tensor, nn.Parameter,int]=32,
+        ny: Union[torch.Tensor, nn.Parameter,int]=32,
+        ns: Union[torch.Tensor, nn.Parameter,int]=32,
+        dx: Union[torch.Tensor, nn.Parameter]=3,
+        dy: Union[torch.Tensor, nn.Parameter]=3,
+        ds: Union[torch.Tensor, nn.Parameter]=3,
```

Comment:
Good question. I think their role is very similar to that of `Screen.resolution`. So I think no one will every want get the gradient with respect to any of them (?), meaning they don't have to be `nn.Parameter`. I think we could get away with making them `int` and `float`. But, in NumPy, if you interleave operations on NumPy types and Python types, computations slow down significantly. I don't know to what extend this is also an issue in PyTorch, but just to avoid it, I would make them all `torch.Tensor`.